### PR TITLE
NAS-117216 / 22.12 / Pools dashboard cards scalling and ordering improved

### PR DESCRIPTION
**Pull request by AlexKarpov98**

### CODE REVIEW

File: src/app/pages/datasets/components/dataset-management/dataset-management.component.ts [hunk 1]
```diff
@@ -86,11 +87,22 @@ export class DatasetsManagementComponent implements OnInit {
...
+    this.dataSource = new IxNestedTreeDataSource<DatasetInTree>(filteredDatasets);
+    this.dataSource.filterPredicate = (filteredDatasets, query = '') => {
+      return filteredDatasets.map((datasetRoot) => {
         return findInTree([datasetRoot], (dataset) => dataset.id.toLowerCase().includes(query.toLowerCase()));
       }).filter(Boolean);
     };
   }
+
+  private initFilterDataSource(datasets: DatasetInTree[]): DatasetInTree[] {
+    const filtered = (datasets: DatasetInTree[]): DatasetInTree[] => {
```
[the remark this answers is not in the document]
Comment (AlexKarpov98) [reply src/app/pages/datasets/components/dataset-management/dataset-management.component.ts]: That was code of another ticket.
Removed that code this branch.

**Review by bvasilenko — APPROVED** [review]
Works good to me!

Jul-20-2022 12-19-16

**Review by RehanY147 — CHANGES_REQUESTED** [review]
I think a better approach would be to spread the content out dynamically. Extra height at the bottom of cards that isn't used looks odd. Also, no margin/padding on the right edge.
image

**Comment by RehanY147** [comment]
We can figure out the tallest card. Fix that height and then make all the other cards match that height by spreading the content out with margins.

**Comment by AlexKarpov98** [comment]
> I think a better approach would be to spread the content out dynamically. Extra height at the bottom of cards that isn't used looks odd. Also, no margin/padding on the right edge. image


The  "no margin/padding on the right edge" issue was fixed.
Please check for the last update agian.

https://github.com/truenas/webui/pull/6879/commits/e84ecc6125f4ec4c7d5aa49cc0a143317d037ce8#diff-8a22545de0ceef327f73ff7d5c5c20e4b846a1b5316318bcc3164097da03b892

**Comment by AlexKarpov98** [comment]
> We can figure out the tallest card. Fix that height and then make all the other cards match that height by spreading the content out with margins. All of these cards are going to be static in vertical height. So, we can fix their margins without worrying about stuff changing.

If all the cards are arranged with a row, that's possible definitely but, I am not sure if the cards are arranged with 2 rows the height of all the cards will be fixed as the same size by the css.

**Comment by RehanY147** [comment]
@AlexKarpov98 I didn't mean do it dynamically. I meant that let's figure out the height of the tallest card once. Let's say it's 500px. Let's set the height of all cards to 500px. Then spread the rows out so there isn't much space at the bottom left out. Let me know if you want to discuss this further.

**Comment by RehanY147** [comment]
If there's too little to show, like in the ZFS health card in your screenshot, there's only three rows, we can spread them out at least a little bit and then leave some space at the bottom. Just make it less obvious that we're forcing cards to have the same height. Doesn't have to be perfect. Just a little bit better.

**Review by RehanY147 — COMMENTED** [review]
^

**Comment by AlexKarpov98** [comment]
Removed fixed height of the cards.
All the cards on the same row takes the same height with an height of the highest card of them.

**Review by RehanY147 — COMMENTED** [review]
Discussed on Slack

**Comment by AlexKarpov98** [comment]
> Rows need more vertical margins in the ZFS card.

Displayed as spread row.

image

**Review by RehanY147 — CHANGES_REQUESTED** [review]
As discussed on slack, let's standardize the margins between rows on all cards similar to how they were previously on the topology card.

**Comment by AlexKarpov98** [comment]
> As discussed on slack, let's standardize the margins between rows on all cards similar to how they were previously on the topology card.

Standardized a vertical margin between row as 24px on all the cards.
image

File: src/app/pages/storage2/components/zfs-health-card/zfs-health-card.component.scss [hunk 2]
```diff
@@ -14,7 +14,7 @@
   }
 
   .details-item {
-    margin: 0;
+    margin: 12px 0;
```
Comment (RehanY147) [inline src/app/pages/storage2/components/zfs-health-card/zfs-health-card.component.scss:17]: I think `8px` would be a better fit

File: src/app/pages/storage2/components/pools-dashboard/widget-topology/widget-topology.component.html [hunk 3]
```diff
@@ -22,7 +22,7 @@ <h3 mat-card-title>
   </mat-card-header>
 
   <mat-card-content fxFlex="100">
-    <div class="content-container" fxLayout="column" fxLayoutAlign="space-between">
+    <div class="content-container" fxLayout="column">
```
Comment (RehanY147) [inline src/app/pages/storage2/components/pools-dashboard/widget-topology/widget-topology.component.html:25]: Let's switch the `<div>` underneath with `*ngIf="!loading"` with `ng-container` like so
`<ng-container *ngIf="!loading" fxLayout="row" ...>`
Also bring back the `fxLayoutAlign="space-between"` here to fix the weird alignment of the `Data VDEVs` row.

File: src/app/pages/storage2/components/disk-health-card/disk-health-card.component.scss [hunk 4]
```diff
@@ -14,15 +14,8 @@
...
-  .mat-card-content {
-    display: flex;
-    flex: 1;
-    flex-direction: column;
-    justify-content: space-between;
-  }
-
   .details-item {
-    margin: 0;
+    margin: 12px 0;
```
Comment (RehanY147) [inline src/app/pages/storage2/components/disk-health-card/disk-health-card.component.scss:18]: `8px` here too

**Review by RehanY147 — CHANGES_REQUESTED** [review]

File: src/app/pages/storage2/components/zfs-health-card/zfs-health-card.component.scss (hunk 2 above)
Comment (AlexKarpov98) [reply src/app/pages/storage2/components/zfs-health-card/zfs-health-card.component.scss]: Done

File: src/app/pages/storage2/components/disk-health-card/disk-health-card.component.scss (hunk 4 above)
Comment (AlexKarpov98) [reply src/app/pages/storage2/components/disk-health-card/disk-health-card.component.scss]: Done

File: src/app/pages/storage2/components/pools-dashboard/widget-topology/widget-topology.component.html (hunk 3 above)
Comment (AlexKarpov98) [reply src/app/pages/storage2/components/pools-dashboard/widget-topology/widget-topology.component.html]: Added `<ng-container *ngIf="!loading">...</ng-container>`.
image

A `fxLayoutAlign="space-between"` attribute was removed already.

**Review by RehanY147 — CHANGES_REQUESTED** [review]
Data VDEVs still is in the middle of the card unlike the other rows. Was this an intentional change? Because I think it's coming from master. That's why I was asking fxLayoutAlign="space-between" to be brought back.
image

**Comment by AlexKarpov98** [comment]
> Data VDEVs still is in the middle of the card unlike the other rows. Was this an intentional change? Because I think it's coming from master. That's why I was asking fxLayoutAlign="space-between" to be brought back. image



> Data VDEVs still is in the middle of the card unlike the other rows. Was this an intentional change? Because I think it's coming from master. That's why I was asking fxLayoutAlign="space-between" to be brought back. image

That wasn't an  intentional change.
Maybe that has been removed as a mistake before previous PR, I think.
But that attribute should be removed in a `content-container` div.
Instead, the lines should be arranged by `fxLayoutAlign="space-between"` attribute in a `vdev-line` div.
That was fixed by this PR https://github.com/truenas/webui/pull/6917.

This is the code fixed.
image

**Review by RehanY147 — COMMENTED** [review]
The build is failing. Also, I'm seeing 0 files changed. What's that about? Do you know?

**Comment by RehanY147** [comment]
Nothing left to merge.

**Pull request closed without merging**